docs/source/user-guide/sql/ddl.md

```diff
@@ -30,9 +30,17 @@ STORED AS PARQUET
 LOCATION '/mnt/nyctaxi/tripdata.parquet';
 ```
 
-CSV data sources can also be registered by executing a `CREATE EXTERNAL TABLE` SQL statement. It is necessary to
-provide schema information for CSV files since DataFusion does not automatically infer the schema when using SQL
-to query CSV files.
+CSV data sources can also be registered by executing a `CREATE EXTERNAL TABLE` SQL statement. The schema will be
+inferred based on scanning a subset of the file.
+
+```sql
+CREATE EXTERNAL TABLE test
+STORED AS CSV
+WITH HEADER ROW
+LOCATION '/path/to/aggregate_test_100.csv';
```

Comment:
I didn't realize you could do this 🤔 

I actually tried this particular example and it doesn't work: 

```
❯ CREATE EXTERNAL TABLE test
STORED AS CSV
WITH HEADER ROW
LOCATION '../testing/data/csv/aggregate_test_100.csv';
0 rows in set. Query took 0.102 seconds.
❯ select * from test;
ArrowError(ParseError("Error while parsing value 14857091259186476033 for column 9 at line 3"))

```

Reply:
```suggestion
LOCATION '/path/to/aggregate_simple.csv';
```

But an alternate worked great: 

```sql
❯ CREATE EXTERNAL TABLE test
STORED AS CSV
WITH HEADER ROW
LOCATION '../datafusion/core/tests/aggregate_simple.csv';
0 rows in set. Query took 0.014 seconds.
❯ select * from test;
+---------+----------------+-------+
| c1      | c2             | c3    |
+---------+----------------+-------+
| 0.00001 | 0.000000000001 | true  |
| 0.00002 | 0.000000000002 | false |
| 0.00002 | 0.000000000002 | false |
| 0.00003 | 0.000000000003 | true  |
| 0.00003 | 0.000000000003 | true  |
| 0.00003 | 0.000000000003 | true  |
| 0.00004 | 0.000000000004 | false |
| 0.00004 | 0.000000000004 | false |
| 0.00004 | 0.000000000004 | false |
| 0.00004 | 0.000000000004 | false |
| 0.00005 | 0.000000000005 | true  |
| 0.00005 | 0.000000000005 | true  |
| 0.00005 | 0.000000000005 | true  |
| 0.00005 | 0.000000000005 | true  |
| 0.00005 | 0.000000000005 | true  |
+---------+----------------+-------+
```

Reply:
Thanks for testing that. Filed https://github.com/apache/arrow-datafusion/issues/3174